src/sentry/integrations/github/repository.py

```diff
@@ -56,22 +56,32 @@ def build_repository_config(self, organization, data):
         }
 
     def compare_commits(self, repo, start_sha, end_sha):
+        def eval_commits(client):
+            # use config name because that is kept in sync via webhooks
+            name = repo.config['name']
+            if start_sha is None:
+                res = client.get_last_commits(name, end_sha)
+                return self._format_commits(client, name, res[:20])
+            else:
+                res = client.compare_commits(name, start_sha, end_sha)
+                return self._format_commits(client, name, res['commits'])
+
         integration_id = repo.integration_id
         if integration_id is None:
             raise NotImplementedError('GitHub apps requires an integration id to fetch commits')
         integration = Integration.objects.get(id=integration_id)
         installation = integration.get_installation(repo.organization_id)
         client = installation.get_client()
 
-        # use config name because that is kept in sync via webhooks
-        name = repo.config['name']
         try:
-            if start_sha is None:
-                res = client.get_last_commits(name, end_sha)
-                return self._format_commits(client, name, res[:20])
-            else:
-                res = client.compare_commits(name, start_sha, end_sha)
-                return self._format_commits(client, name, res['commits'])
+            return eval_commits(client)
+        except ApiError as e:
+            if '404' in e.message:
```

Comment:
Are you able to use `e.code` for checking the status code?

Reply:
Oh you're totally right! Thanks for the reminder :)